### PR TITLE
Make 3D Triangulate possible with pytorch_dlc branch and MA project

### PR DESCRIPTION
**Pull request by Zelin2001**
## Rationality

I proceeded with my project, and then found the `triangulate()` function dependent on TensorFlow modules. Further debugging revealed more problems relating to the 'single' individual, who has a different bodypart dimension from common individuals.

Finally, I made minor changes to make triangulation work on my dataset. These changes might also help with other 3D MA users.

## Changes

- Compute triangulated data with pose_estimation_pytorch

- Make correct MultiIndex with unique bodyparts

- adapt to DataFrame.to_hdf() changes

### CODE REVIEW

**Comment by MMathisLab** [comment]
thanks so much for your contribution! This is super helpful!

**Review by MMathisLab — APPROVED** [review]

File: deeplabcut/pose_estimation_3d/triangulation.py [hunk 1]
```diff
@@ -15,13 +15,14 @@
 import cv2
 import numpy as np
 import pandas as pd
-from matplotlib.axes._axes import _log as matplotlib_axes_logger
+
+# from matplotlib.axes._axes import _log as matplotlib_axes_logger
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_3d/triangulation.py]: If we want to change the logging level I'd prefer removing the lines entirely rather than commenting them :)

```suggestion
```

File: deeplabcut/pose_estimation_3d/triangulation.py [hunk 2]
```diff
@@ -15,13 +15,14 @@
...
 
 from deeplabcut.utils import auxfun_multianimal, auxiliaryfunctions
 from deeplabcut.utils import auxiliaryfunctions_3d
 from deeplabcut.core.trackingutils import TRACK_METHODS
 
-matplotlib_axes_logger.setLevel("ERROR")
+# matplotlib_axes_logger.setLevel("ERROR")
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_3d/triangulation.py]: same comment as above :)

```suggestion
```

File: deeplabcut/pose_estimation_3d/triangulation.py [hunk 3]
```diff
@@ -88,7 +89,7 @@ def triangulate(
     To analyze only a few pair of videos:
     >>> deeplabcut.triangulate(config,[['C:\\yourusername\\rig-95\\Videos\\video1-camera-1.avi','C:\\yourusername\\rig-95\\Videos\\video1-camera-2.avi'],['C:\\yourusername\\rig-95\\Videos\\video2-camera-1.avi','C:\\yourusername\\rig-95\\Videos\\video2-camera-2.avi']])
     """
-    from deeplabcut.pose_estimation_tensorflow import predict_videos
+    from deeplabcut.pose_estimation_pytorch import analyze_videos
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_3d/triangulation.py]: I believe this should be updated to import analyze_videos from the `compat` module. This will make the code compatible with both `PyTorch` and `TensorFlow` DeepLabCut models, as it will select which to use based on the shuffle/which type of model was trained

```suggestion
    from deeplabcut.compat import analyze_videos
```

File: deeplabcut/pose_estimation_3d/triangulation.py [hunk 4]
```diff
@@ -263,13 +264,12 @@ def triangulate(
...
                     else:
                         # Analyze video if score name is different
-                        DLCscorer = predict_videos.analyze_videos(
+                        DLCscorer = analyze_videos(
                             config_2d,
                             [video],
                             videotype=videotype,
                             shuffle=shuffle,
                             trainingsetindex=trainingsetindex,
-                            gputouse=gputouse,
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_3d/triangulation.py]: If we import `analyze_videos` from the `compat` module as described above, we can keep the `gputouse` parameter as it will be handled correctly (see this [bit of code](https://github.com/DeepLabCut/DeepLabCut/blob/78cc44eaea296fc2a6f8979027497298348359aa/deeplabcut/compat.py#L849)).

This will allow users to continue choosing on which GPU they want to run video analysis :)

File: deeplabcut/pose_estimation_3d/triangulation.py [hunk 5]
```diff
@@ -293,13 +293,12 @@ def triangulate(
...
 
                 else:  # need to do the whole jam.
-                    DLCscorer = predict_videos.analyze_videos(
+                    DLCscorer = analyze_videos(
                         config_2d,
                         [video],
                         videotype=videotype,
                         shuffle=shuffle,
                         trainingsetindex=trainingsetindex,
-                        gputouse=gputouse,
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_3d/triangulation.py]: Same comment as above :)

**Review by n-poulsen — CHANGES_REQUESTED** [review]
Thanks @Zelin2001, this looks great! Just suggested a few changes, let me know if you want to discuss anything.

One the changes are made this looks good to me :)

**Comment by Zelin2001** [comment]
> Thanks @Zelin2001, this looks great! Just suggested a few changes, let me know if you want to discuss anything.
> 
> One the changes are made this looks good to me :)

Thank you a lot for the advice @n-poulsen . I've learned more about deeplabcut from these changes.

**Review by n-poulsen — APPROVED** [review]
Looks good to me! Thanks @Zelin2001!
